src/lib/containers/table/SynapseTable.tsx

```diff
@@ -167,21 +167,32 @@ export default class SynapseTable extends React.Component<
     this.getEntityHeadersInData()
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps: QueryWrapperChildProps & SynapseTableProps) {
     this.getEntityHeadersInData()
+    this.getTableConcreteType(prevProps)
+  }
+
+  public async getTableConcreteType(
+    prevProps: QueryWrapperChildProps & SynapseTableProps,
+  ) {
+    const { data, token } = this.props
+    const previousTableId =
+      prevProps.data && prevProps.data.queryResult.queryResults!.tableId
```

Comment:
Invitation for discussion: _'using _.get vs checking for undefined'_
we do a lot of this :
`const previousTableId = prevProps.data && prevProps.data.queryResult.queryResults!.tableId` throughout the codebase which can be instead accomplished with 
`const previousTableId = _.get(prevProps, 'data.queryResult.queryResults.tableId')`

I think that latter approach makes the code cleaner but you are loosing the type safety of TS.

Here is an interesting alternative:
https://codewithstyle.info/Deep-property-access-in-TypeScript/
custom util function?